mlflow/sklearn/__init__.py

```diff
@@ -1787,8 +1787,13 @@ def out(*args, **kwargs):
                 flavor_name, sklearn.metrics, metric_name, patched_metric_api, manage_run=False
             )
 
-        for scorer in sklearn.metrics.SCORERS.values():
-            safe_patch(flavor_name, scorer, "_score_func", patched_metric_api, manage_run=False)
+        if Version(sklearn.__version__) > Version("1.2.2"):
+            for scoring in sklearn.metrics.get_scorer_names():
+                scorer = sklearn.metrics.get_scorer(scoring)
+                safe_patch(flavor_name, scorer, "_score_func", patched_metric_api, manage_run=False)
+        else:
+            for scorer in sklearn.metrics.SCORERS.values():
+                safe_patch(flavor_name, scorer, "_score_func", patched_metric_api, manage_run=False)
```

Comment:
you don't need to duplicate the for-loop body code,

you can write:
```
if (Version(sklearn.__version__) > Version("1.2.2")
  sklearn_scorer_names = ...
else:
  sklearn_scorer_names = ...

for scorer in sklearn_scorer_names:
  ...
```

Reply:
We're iterating over the scorers not the names. If iterating over the names, we need to either use 'sklearn.metrics.get_scorer(name)' to fetch it in version >1.2.2 or 'SCORERS[name]' for versions <=1.2.2

Reply:
oh I miss it.